### DISCUSSION TITLE
Disable single option (radio button) from option-picker based on its caption

### DISCUSSION THREAD

**Discussion by pandel** [Budibase/budibase#7462] [Help] [answered]
Hi!

Is it possible to disable a single option (radio button) from an option-picker based on the caption of the radio button itself?

The option-picker gets its values from the schema.

Let's say, the option picker has 4 values to choose from:
- red
- green
- blue
- white

Now, in some conditions, I'd like to deactivate the radio button for 'blue', but leave `red`, `green`, and `white` active while also showing `blue`(deactivated) to the user.

Regards,
Holger

**Comment by R2bEEaton** [comment, marked as answer]
Hi @pandel,

There is currently no way to achieve exactly what you want. I see two options.

### Filter
You could use something like a filter if you get options from a data source. You can see an example of this in #7449. Instead of disabling blue, you could take it away from the list entirely.

### Validation
Another option is using Conditions to update the Validation setting to something like "Must not match Regex" `(blue|red)` which would make it so it warns the user that they can't select blue or red if they do. Make sure that if you have a submit button, the first step is to trigger validation for the form so it doesn't let the user continue anyway. The validation doesn't need to be in a condition either, you could set a normal validation and have the Regex be a Handlebar like `({{ join COLORLIST "|" }})` which would build a Regex string based on the contents of `COLORLIST`.

Hope this helps! Reply with any questions or

**Reply by pandel** [reply]
@R2bEEaton 
Many thanks! Sorry for the late reply...

I understand. Filtering is not the best option, as the user should see 'blue'. So validation is the better way to go. And this regex handlebar thing seems reasonable. I will try that!

**Reply by R2bEEaton** [reply]
Perfect, good luck!
